CONTRIBUTING.md

```diff
@@ -1,76 +1,262 @@
+{
+  CGFloat contentSpacing = 0.0;
+  __instanceLock__.lock();
+  contentSpacing = _contentSpacing;
+  __instanceLock__.unlock();
+  return contentSpacing;
+}
+```
+  2. Create an `ASDN::MutexLocker` :
+```objc
+- (void)setContentSpacing:(CGFloat)contentSpacing
+{
+  {
+    ASDN::MutexLocker l(__instanceLock__);
+    if (contentSpacing == _contentSpacing) {
+      return;
+    }
+
+    _contentSpacing = contentSpacing;
+  }
+
+  [self setNeedsLayout];
+}
+
+- (CGFloat)contentSpacing
+{
+  ASDN::MutexLocker l(__instanceLock__);
+  return _contentSpacing;
+}
+```
+- Nullability
+  - The adoption of annotations is straightforward. The standard we adopt is using the `NS_ASSUME_NONNULL_BEGIN` and `NS_ASSUME_NONNULL_END` on all headers. Then indicate nullability for the pointers that can be so.
+  - There is mostly no sense using nullability annotations outside of interface declarations.
+```objc
+// Properties
+@property(nonatomic, strong, nullable) NSNumber *status
 
-Complete your CLA here: <https://code.facebook.com/cla>
+// Methods
+- (nullable NSNumber *)doSomethingWithString:(nullable NSString *)str;
+
+// Functions
+NSString * _Nullable ASStringWithQuotesIfMultiword(NSString * _Nullable string);
+
+// Typedefs
+typedef void (^RemoteCallback)(id _Nullable result, NSError * _Nullable error);
+
+// Block as parameter
+- (void)reloadDataWithCompletion:(void (^ _Nullable)())completion;
+
+// Block as parameter with parameter and return value
+- (void)convertObject:(id _Nonnull (^ _Nullable)(id _Nullable input))handler;
+
+// More complex pointer types
+- (void)allElementsForScrolling:(ASScrollDirection)scrollDirection rangeMode:(ASLayoutRangeMode)rangeMode displaySet:(NSSet<ASCollectionElement *> *__autoreleasing  _Nullable *)displaySet preloadSet:(NSSet<ASCollectionElement *> *__autoreleasing  _Nullable *)preloadSet map:(ASElementMap *)map;
+```
 
-## Issues
-We use GitHub issues to track public bugs. Please ensure your description is
-clear and has sufficient instructions to be able to reproduce the issue.
+# Contributor License Agreement (CLA)
 
-Facebook has a [bounty program](https://www.facebook.com/whitehat/) for the safe
-disclosure of security bugs. In those cases, please go through the process
-outlined on that page and do not file a public issue.
+Please sign our Contributor License Agreement (CLA) before sending pull requests. For any code changes to be accepted, the CLA must be signed.
 
-## Getting Help
-We use Slack for real-time debugging, community updates, and general talk about ASDK. Signup at http://asdk-slack-auto-invite.herokuapp.com or email AsyncDisplayKit(at)gmail.com to get an invite.
+Complete your CLA here
 
-## Coding Style
-* 2 spaces for indentation rather than tabs
+# License
 
-## License
-By contributing to AsyncDisplayKit, you agree that your contributions will be
-licensed under its BSD license.
+By contributing to Texture, you agree that your contributions will be licensed under its Apache license.
```

Comment:
Apache 2. You can also say, see README.md for more details (I'll be updating it).